### PR TITLE
fix: remove bad check-fail in the transaction code

### PR DESCRIPTION
**Pull request by romange**
Required to track errors easily with `--vmodule=reply_builder=1`

### CODE REVIEW

**Review by dranikpg — APPROVED** [review]

File: src/server/transaction.h [hunk 1]
```diff
@@ -329,16 +329,16 @@ class Transaction {
     char pad[46];  // to make sure PerShardData is 64 bytes and takes full cacheline.
 
     uint32_t arg_start = 0;  // Indices into args_ array.
-    uint16_t arg_count = 0;
-
-    // Accessed within shard thread.
-    // Bitmask of LocalState enums.
-    uint16_t local_mask = 0;
+    uint32_t arg_count = 0;
```
Comment (dranikpg) [inline src/server/transaction.h:332]: I hope you grepped that uint16_t isn't used anywhere else 🙂

File: src/server/transaction.cc [hunk 2]
```diff
@@ -157,7 +155,7 @@ void Transaction::InitShardData(absl::Span<const PerShardCache> shard_index, siz
     }
   }
 
-  CHECK(args_.size() == num_args);
+  CHECK_EQ(args_.size(), num_args);
```
Comment (ashotland) [inline src/server/transaction.cc:158]: The PR title, description and the CHECK that is still here is not very coherent to me.
Consider clarifying.

Comment (romange) [reply src/server/transaction.cc:158]: this is a different check-fail

File: src/server/transaction.cc [hunk 3]
```diff
@@ -133,8 +133,6 @@ void Transaction::InitShardData(absl::Span<const PerShardCache> shard_index, siz
     auto& sd = shard_data_[i];
     auto& si = shard_index[i];
 
-    CHECK_LT(si.args.size(), 1u << 15);
```
Comment (romange) [inline src/server/transaction.cc:136]: @ashotland  this is what has failed.

File: src/server/transaction.h (hunk 1 above)
Comment (romange) [reply src/server/transaction.h:332]: eeh, now i did :)

**Pull request merged**